testsuite/MDAnalysisTests/analysis/test_rdf.py

```diff
@@ -94,6 +96,24 @@ def test_exclusion(sels):
     assert rdf.results.count.sum() == 4
 
 
+@pytest.mark.parametrize("attr, count", [
+    ("residue", 8),
+    ("segment", 0),
+    ("chain", 8)])
+def test_ignore_same_residues(sels, attr, count):
+    # should see two distances with 4 counts each
+    s1, s2 = sels
+    rdf = InterRDF(s2, s2, exclude_same=attr).run()
+    assert rdf.rdf[0] == 0
+    assert rdf.results.count.sum() == count
+
+
+def test_ignore_same_residues_fails(sels):
+    # should see two distances with 4 counts each
+    s1, s2 = sels
+    with pytest.raises(ValueError, match="The exclude_same argument to InterRDF must be"):
+        rdf = InterRDF(s2, s2, exclude_same="unsupported").run()
+
```

Comment:
```suggestion

    with pytest.raises(ValueError, match="The exclude_same argument to InterRDF cannot be used with"):
        rdf = InterRDF(s2, s2, exclude_same="unsupported", exclude_block=tuple()).run()
        
        
```
Last nitpick, something like this should get ride of code coverage complaints.

Reply:
@orionarcher Sorry, this was mainly an (untested) suggestion. `exclude_same` needs to be changed to a valid entry in order not to trigger the previous exception.

Reply:
Sorry about that, fixed now! Hopefully testing passes.